spawn/specification/evaluators.py

```diff
@@ -62,7 +62,11 @@ def evaluate(self, **kwargs):
 
     #pylint: disable=no-self-use
     def _evaluate_arg(self, arg, **kwargs):
-        return evaluate(arg, **kwargs) if isinstance(arg, ValueProxy) else arg
+        try:
+            return evaluate(arg, **kwargs) if isinstance(arg, ValueProxy) else arg
+        except ParameterNotFoundError as ex:
+            evaluator_name = self._name or type(self).__name__
+            raise EvaluatorParameterNotFoundError(ex.parameter_name, evaluator_name)
```

Comment:
You can use the syntax `raise new_exception from original_exception` - this makes things a bit clearer in the trace

Reply:
ok - implemented